implementations/rust/ockam/ockam_identity/src/secure_channel/encryptor_worker.rs

```diff
@@ -66,9 +67,34 @@ impl EncryptorWorker {
     }
 
     /// Encrypt the message
-    async fn encrypt(&mut self, ctx: &Context, msg: SecureChannelMessage) -> Result<Vec<u8>> {
-        match self.encryptor.encrypt(&minicbor::to_vec(&msg)?).await {
-            Ok(encrypted_payload) => Ok(encrypted_payload),
+    async fn encrypt(&mut self, ctx: &Context, msg: SecureChannelMessage<'_>) -> Result<Vec<u8>> {
```

Comment:
What is the difference between `encrypt` and `encrypt_to`? Aren't they doing the same thing?

Reply:
Yes they are, one takes a writable `Vec` argument the other returns it. Could delete the returning one and force every caller to create the destination beforehand, but it would make the code a bit less readable.

Reply:
Why not at least call `encrypt_to` from `encrypt`?